EEJT-000A/main/java/task2/SearchByDateChange.java

```diff
@@ -0,0 +1,65 @@
+package task2;
+
+import java.io.File;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class SearchByDateChange implements SearchByParam {
+
+    private SearchByParam searchByParam;
+    private final String Patt = "yyyy-MM-dd HH:mm:ss";
+
+    public SearchByDateChange(SearchByParam searchByParam) {
+        this.searchByParam = searchByParam;
+    }
+    private boolean isNextChain (SearchByParam searchByParam){
+        return searchByParam != null;
```

Comment:
Judging by the logic I think this method should be named `hasNextChain` or `hasNextChainElement`